mesonbuild/build.py

```diff
@@ -1800,8 +1800,11 @@ def get_arglist(self, inname: str) -> T.List[str]:
 
     @staticmethod
     def is_parent_path(parent: str, trial: str) -> bool:
-        relpath = pathlib.PurePath(trial).relative_to(parent)
-        return relpath.parts[0] != '..' # For subdirs we can only go "down".
+        try:
+            common = os.path.commonpath((parent, trial))
+        except ValueError: # Windows on different drives
+            return False
+        return pathlib.PurePath(common) == pathlib.PurePath(parent)
```

Comment:
Despite your hateness for pathlib, I still argue that the right thing to do is:

```suggestion
        try:
            pathlib.PurePath(trial).relative_to(parent)
            return True
        except ValueError: # Windows on different drives
            return False
```